src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1099,28 +1099,31 @@ private bool IsValidFixedVariableInitializer(TypeSymbol declType, SourceLocalSym
                     if (initializerType.SpecialType == SpecialType.System_String &&
                         ((object)fixedPatternMethod == null || fixedPatternMethod.ContainingType.SpecialType != SpecialType.System_String))
                     {
+                        fixedPatternMethod = null;
```

Comment:
Don't we still want this to be null if we have we're not inside System.String and we call GetFixedLocalCollectionInitializer with fixedPatternMethod below?

Reply:
That was actually a bug, because if GetFixedLocalCollectionInitializer  gets nonnull method, it will assume that is the method to use. By setting to null we are signaling that regular System.String pattern should be used.

Without this change, for example, an extension method on String would win over the pattern. We would not want that. (see: `FixStringArrayExtensionHelpersIgnored`)

---
In reply to: [176875972](https://github.com/dotnet/roslyn/pull/25660#discussion_r176875972) [](ancestors = 176875972)